modules/apps/static/portal-k8s-agent/portal-k8s-agent-impl/src/main/java/com/liferay/portal/k8s/agent/internal/LiferayRoutesEmitter.java

```diff
@@ -0,0 +1,507 @@
+				ModelListener.class, new CompanyModelListener(), null));
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		for (ServiceRegistration<?> serviceRegistration :
+				_serviceRegistrations) {
+
+			serviceRegistration.unregister();
+		}
+	}
+
+	private void _deleteFolder(Path folderPath) throws Exception {
+		if (!Files.exists(folderPath)) {
+			return;
+		}
+
+		Files.walkFileTree(
+			folderPath,
+			new SimpleFileVisitor<Path>() {
+
+				@Override
+				public FileVisitResult postVisitDirectory(
+						Path dirPath, IOException ioException)
+					throws IOException {
+
+					if (ioException != null) {
+						throw ioException;
+					}
+
+					Files.delete(dirPath);
+
+					return FileVisitResult.CONTINUE;
+				}
+
+				@Override
+				public FileVisitResult visitFile(
+						Path filePath, BasicFileAttributes basicFileAttributes)
+					throws IOException {
+
+					Files.delete(filePath);
+
+					return FileVisitResult.CONTINUE;
+				}
+
+			});
+	}
+
+	private void _deleteLiferayRoutes(
+			String configMapName, Map<String, String> labels)
+		throws Exception {
+
+		Path liferayRoutesPath = _getLiferayRoutesPath();
+
+		String virtualInstanceId = labels.get(
+			"dxp.lxc.liferay.com/virtualInstanceId");
+
+		if ((liferayRoutesPath == null) || (virtualInstanceId == null)) {
+			return;
+		}
+
+		if (Objects.equals(
+				virtualInstanceId, PropsValues.COMPANY_DEFAULT_WEB_ID)) {
+
+			virtualInstanceId = "default";
+		}
+
+		Path virtualInstanceIdPath = liferayRoutesPath.resolve(
+			virtualInstanceId);
+
+		Matcher matcher = _lxcDxpMetadataPattern.matcher(configMapName);
+
+		if (matcher.matches()) {
+			if (Files.exists(virtualInstanceIdPath)) {
+				_deleteFolder(virtualInstanceIdPath);
+			}
+		}
+		else {
+			matcher = _lxcExtInitMetadataPattern.matcher(configMapName);
+
+			String projectName = labels.get("ext.lxc.liferay.com/projectName");
+
+			if (matcher.matches() && (projectName != null)) {
+				Path projectPath = virtualInstanceIdPath.resolve(projectName);
+
+				if (Files.exists(projectPath)) {
+					_deleteFolder(projectPath);
+				}
+			}
+		}
+	}
+
+	private Path _getLiferayRoutesPath() {
+		String liferayHome = PropsValues.LIFERAY_HOME;
+
+		if (!FileUtil.exists(liferayHome)) {
+			return null;
+		}
+
+		Path liferayRoutesPath = Paths.get(
+			liferayHome, PropsValues.LIFERAY_ROUTES);
+
+		try {
+			liferayRoutesPath = Files.createDirectories(liferayRoutesPath);
+		}
+		catch (IOException ioException) {
+			_log.error("Could not create Liferay routes path", ioException);
+		}
+
+		return liferayRoutesPath;
+	}
+
+	private int _getPortalLocalPort() {
+		return _portal.getPortalLocalPort(
+			Objects.equals(_getWebServerProtocol(), "https"));
+	}
+
+	private String _getWebServerProtocol() {
+		String webServerProtocol = PropsValues.WEB_SERVER_PROTOCOL;
+
+		if (Validator.isNull(webServerProtocol)) {
+			return Http.HTTP;
+		}
+
+		return webServerProtocol;
+	}
+
+	private void _updateDxpRoutes(boolean secure) {
```

Comment:
here you have `dxproutes`

Reply:
My thinking was this is only updating the 'dxp' routes